### PR TITLE
feature/optimize_handle_accounting_metadata

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Optimizations found after profiling handle accounting metadata

Enhancements
-------------
* Optimized node lookup
* Optimized API client creation
* Optimized logging usage
* Optimized mac clean

Delete branch after merge
=========================
NO

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

**Comment by julsemaan** [comment]
I checked it out and it looks good.

Have you also performed the profiling using the local API client so that we can tune the stuff in radius.pm ?

**Comment by jrouzierinverse** [comment]
Yes, nothing that I could optimize easily.

**Comment by julsemaan** [comment]
Can you post performance before and after (using local client)

Its not because it won't be optimizable easily that we shouldn't do it. 
The goal is to have extremely scalable accounting for v10 and if we need to rework some of the core of the accounting, we'll have to do so.

File: lib/pf/api.pm [hunk 1]
```diff
@@ -1475,51 +1443,54 @@ sub radius_rest_accounting :Public :RestPath(/radius/rest/accounting) {
...
-    if ($RAD_REQUEST{'Acct-Status-Type'} != $ACCOUNTING::STOP){
+    if ($acct_status_type != $ACCOUNTING::STOP){
+        my $advanced = $pf::config::Config{advanced};
         # Tracking IP address.
-        if(pf::util::isenabled($pf::config::Config{advanced}{update_iplog_with_accounting})){
+        my $framed_ip = $RAD_REQUEST->{"Framed-IP-Address"};
+        if ($framed_ip && pf::util::isenabled($advanced->{update_iplog_with_accounting})) {
             $logger->info("Updating iplog from accounting request");
-            $client->notify("update_ip4log", mac => $mac, ip => $RAD_REQUEST{'Framed-IP-Address'}) if ($RAD_REQUEST{'Framed-IP-Address'} );
+            $client->notify("update_ip4log", mac => $mac, ip => $framed_ip);
```
Comment (fdurand) [inline lib/pf/api.pm:1467]: To get rid of a call , can we test if the $framed_ip is a valid ip ?

File: lib/pf/api.pm [hunk 2]
```diff
@@ -1475,51 +1443,54 @@ sub radius_rest_accounting :Public :RestPath(/radius/rest/accounting) {
...
         }
         else {
-            pf::log::get_logger->debug("Not handling iplog update because we're not configured to do so on accounting packets.");
+            $logger->debug("Not handling iplog update because we're not configured to do so on accounting packets.");
         }
         
-        if(pf::util::isenabled($pf::config::Config{advanced}{scan_on_accounting}) && $RAD_REQUEST{"Framed-IP-Address"}) {
+        if ($framed_ip && pf::util::isenabled($advanced->{scan_on_accounting})) {
```
Comment (fdurand) [inline lib/pf/api.pm:1473]: if (valid_ip($framed_ip) && ...

**Pull request closed without merging**